frontend/src/lib/components/PropertyFilters/components/PropertyValue.tsx

```diff
@@ -97,25 +98,26 @@ export function PropertyValue({
             return
         }
         const key = propertyKey.split('__')[0]
-        setOptions({ ...options, [propertyKey]: { ...options[propertyKey], status: 'loading' } })
+        setOptions({ ...options, [propertyKey]: { ...options[propertyKey] } })
+        setLoading(true)
```

Comment:
I'd revert the `loading` pattern here, or you'll introduce a subtle bug that's not visible today, but may be tomorrow. Previously the loading state was set per key (`email` vs `$browser`), now it's global to all keys. This can cause issues.

However this component is too complicated for its own good and should probably become a logic. All this API code would sit so much better in a loader.

Reply:
The subtle bug we have on master currently. Is that `loading` is never true :)

<img width="527" alt="Screenshot 2022-01-14 at 13 32 57" src="https://user-images.githubusercontent.com/984817/149523567-a800d88f-5469-4ff4-b856-4cc2ee702ae0.png">

That's me interacting with the property filter component with the browser set to Slow 3G network throttling

I agree that it would be better having a logic

Reply:
I'll push that into a different piece of work though... Try to stick to small PRs

Reply:
You can see in the gif in https://github.com/PostHog/posthog/pull/8055#discussion_r784863396 that the loading state is still not correctly propagated

So, now it only works because the loading state is in the empty state because it never receives false instead of never receiving true :)